provider/azure/internal/azureauth/token_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azureauth_test
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/provider/azure/internal/ad"
+	"github.com/juju/juju/provider/azure/internal/azureauth"
+	"github.com/juju/juju/provider/azure/internal/azuretesting"
+)
+
+type TokenSuite struct {
+	testing.IsolationSuite
+	clock   *testing.Clock
+	newUUID func() (utils.UUID, error)
+}
+
+var _ = gc.Suite(&TokenSuite{})
+
+func (s *TokenSuite) SetUpTest(c *gc.C) {
+	s.IsolationSuite.SetUpTest(c)
+	uuids := []string{
```

Comment:
This seems like it's second-guessing what's going to be run in the tests a bit too much.
I think it might be better have something like:

    // uuidGen returns a newUUID function that returns the given
    // uuids in sequence when called.
    func uuidGen(uuids ...string) func() (utils.UUID, error)

and use that explicitly in the test(s)